pkg/utils/defaulter/clusterresourceplacement.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package defaulter is an interface for setting default values for a resource.
+package defaulter
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+const (
+	// DefaultMaxUnavailableValue is the default value of MaxUnavailable in the rolling update config.
+	DefaultMaxUnavailableValue = "25%"
+
+	// 	DefaultMaxSurgeValue is the default value of MaxSurge in the rolling update config.
+	DefaultMaxSurgeValue = "25%"
+
+	// DefaultUnavailablePeriodSeconds is the default period of time we consider a newly applied workload as unavailable.
+	DefaultUnavailablePeriodSeconds = 60
+
+	// DefaultMaxSkewValue is the default degree to which resources may be unevenly distributed.
+	DefaultMaxSkewValue = 1
+
+	// DefaultRevisionHistoryLimitValue is the default value of RevisionHistoryLimit.
+	DefaultRevisionHistoryLimitValue = 10
+)
+
+// SetDefaultsClusterResourcePlacement sets the default values for ClusterResourcePlacement.
+func SetDefaultsClusterResourcePlacement(obj *fleetv1beta1.ClusterResourcePlacement) {
+	if obj.Spec.Policy == nil {
+		obj.Spec.Policy = &fleetv1beta1.PlacementPolicy{
```

Comment:
the crd has a defaulter

Reply:
It does not hurt though :) otherwise, it may be confusing, some are populated by crd and others are populated here.